cachi2/core/package_managers/cargo/main.py

```diff
@@ -28,13 +64,65 @@ def fetch_cargo_source(request: Request) -> RequestOutput:
     return RequestOutput.from_obj_list(components, environment_variables, project_files)
 
 
-def _resolve_cargo_package(package_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
+def _extract_package_info(path_to_toml: Path) -> dict:
+    # 'value' unwraps the underlying dict and that makes mypy happy (it complains about
+    # mismatching type otherwise despite parsed document having the necessary interface).
+    return tomlkit.parse(path_to_toml.read_text()).value["package"]
```

Comment:
I'm wondering - this likely won't be our sole TOML parsing edeavour, I think it would make sense for us to subclass the parser and add a `to_dict` method wrapping the `value` accessor to make it easier and more predictable for everyone. I simply don't like the commentary about shutting up mypy, yeah, it's that, but at the same time, it's an official interface...
Up to you.

Reply:
I think this is `mypy`'s problem -- the class in question provides correct interface which is supposed to be enough and it is enough from user's perspective, however from `mypy`'s perspective it is not. There is no other necessity for a subclass than placating `mypy` and I find both approaches off-putting. I suggest keeping this as an internal function until we need it somewhere else and then either lift it to `utils` as is or subclass.